### PR TITLE
Automation Tests for Notification and Platform Configuration Managment

### PR DESCRIPTION
**Pull request by Kamidu**

### CODE REVIEW

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            throw new IllegalStateException("This is not the notification-listing page");
+        }
+    }
+
+    public void viewNotification(){
+        WebElement unReadNotificationTab = driver.findElement(By.id(uiElementMapper.
+                getElement("emm.notification.unread.identifier")));
+        WebElement notificationTab = driver.findElement(By.id(uiElementMapper.
+                getElement("emm.notification.all.identifier")));
+
```
Comment (harshanL) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Check formatting of if blocks.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java [hunk 2]
```diff
@@ -50,4 +52,12 @@ public ManagementConsoleLoginPage logout() throws IOException {
         driver.findElement(By.xpath(uiElementMapper.getElement("home.greg.sign.out.xpath"))).click();
         return new ManagementConsoleLoginPage(driver);
     }
+
```
Comment (harshanL) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java]: Check formatting of if blocks.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java (hunk 1 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Recommended fixes  can be found in PR[1]

[1]. https://github.com/wso2/product-mdm/pull/158

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java (hunk 2 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java]: Recommended fixes  can be found in PR[1]

[1]. https://github.com/wso2/product-mdm/pull/158

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        if(!unReadNotificationTab.isDisplayed()){
```
Comment (prabathabey) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Fix formatting issues

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            throw new IllegalStateException("Notification View must be having two tabs");
+        }
+        if(!notificationTab.isDisplayed()){
```
Comment (prabathabey) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Fix formatting issues.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java [hunk 5]
```diff
@@ -50,4 +52,12 @@ public ManagementConsoleLoginPage logout() throws IOException {
...
+    public void checkNotificationCount(String value){
```
Comment (prabathabey) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java]: Fix formatting issues.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java [hunk 6]
```diff
@@ -50,4 +52,12 @@ public ManagementConsoleLoginPage logout() throws IOException {
...
+        WebElement notificationBubble = driver.findElement(By.id(uiElementMapper.
+                getElement("emm.notification.bubble.identifier")));
+        if(!notificationBubble.getText().contains(value)){
```
Comment (prabathabey) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java]: More formatting issues.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    public PlatformConfiguration(WebDriver driver) throws IOException {
+        this.driver = driver;
+        this.uiElementMapper = UIElementMapper.getInstance();
+        // Check that we're on the right page.
+        if (!(driver.getCurrentUrl().contains("platform-configuration"))) {
+            throw new IllegalStateException("This is not the platform-configuration page");
+        }
+    }
+
+    public void changeGeneralConfig(String value){
```
Comment (prabathabey) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java]: Fix formatting issues.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    public void changeAndroidConfig(String value){
+        WebElement androidConfigButton = driver.findElement(By.xpath((uiElementMapper.getElement("emm.configuration.android.tab.identifier"))));
+        WebElement inputAndroidConfig = driver.findElement(By.xpath((uiElementMapper.getElement("emm.configuration.android.input.identifier"))));
+        WebElement buttonSaveConfig = driver.findElement(By.xpath((uiElementMapper.getElement("emm.configuration.android.button.save.identifier"))));
+
+        androidConfigButton.click();
+        inputAndroidConfig.sendKeys(value);
+        buttonSaveConfig.click();
+
```
Comment (prabathabey) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java]: Remove unnecessary new line.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    public void changeWindowsConfig(String value){
+        WebElement windowsConfigButton = driver.findElement(By.xpath((uiElementMapper.getElement("emm.configuration.widows.tab.identifier"))));
+        WebElement inputWindowsConfig = driver.findElement(By.xpath((uiElementMapper.getElement("emm.configuration.windows.input.identifier"))));
+        WebElement buttonSaveConfig = driver.findElement(By.xpath((uiElementMapper.getElement("emm.configuration.windows.button.save.identifier"))));
+
+        windowsConfigButton.click();
+        inputWindowsConfig.sendKeys(value);
+        buttonSaveConfig.click();
+
```
Comment (prabathabey) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java]: Remove unnecessary new line.

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/GeneralConfigurationTestCases.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+package org.wso2.emm.ui.integration.test;
+
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.extensions.selenium.BrowserManager;
+import org.wso2.emm.integration.ui.pages.EMMIntegrationUiBaseTestCase;
+import org.wso2.emm.integration.ui.pages.platformConfiguration.PlatformConfiguration;
+
```
Comment (prabathabey) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/GeneralConfigurationTestCases.java]: Remove unnecessary newlines.

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/NotificationValidationTestCase.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.emm.ui.integration.test;
+
```
Comment (prabathabey) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/NotificationValidationTestCase.java]: Remove unnecessary newline.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java (hunk 5 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java]: These formatting issues has been fixed in a recent PR

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java (hunk 6 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/home/HomePage.java]: These formatting issues has been fixed in a recent PR

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/GeneralConfigurationTestCases.java (hunk 10 above)
Comment (Kamidu) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/GeneralConfigurationTestCases.java]: Recommended fixes can be found in PR[1]

[1]. https://github.com/wso2/product-mdm/pull/166

File: modules/apps/jaggery/mdm/src/mdm/units/notification-listing/notification-listing.hbs [hunk 12]
```diff
@@ -15,13 +15,13 @@
                 <div class="wr-advance-operations">
                     <div class="row no-gutter">
                         <div class="wr-hidden-operations-nav col-lg-4">
-                            <a href="javascript:void(0)" onclick="showAdvanceOperation('unread', this)" class="selected">
+                            <a id=unReadNotifications href="javascript:void(0)" onclick="showAdvanceOperation('unread', this)" class="selected">
```
Comment (dakshika) [inline modules/apps/jaggery/mdm/src/mdm/units/notification-listing/notification-listing.hbs]: unquoted attributes are valid on html5 specification, but as a best practice do recommend to use quotes.

File: modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs [hunk 13]
```diff
@@ -18,7 +18,7 @@
                     {{/each}}
                     ">
                         <div class="wr-hidden-operations-nav col-lg-4">
-                            <a href="javascript:void(0)" onclick="showAdvanceOperation('general', this)" class="selected">
+                            <a id=generalConfigLink href="javascript:void(0)" onclick="showAdvanceOperation('general', this)" class="selected">
```
Comment (dakshika) [inline modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs]: unquoted attributes are valid on html5 specification, but as a best practice do recommend to use quotes.

File: modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs [hunk 14]
```diff
@@ -472,8 +472,8 @@
                     <br>Please click <b>"Go back to configurations"</b>, if you wish to save another configuration or click
                     <b>"Exit"</b> to complete the process and go back to the dashboard.
                     <hr />
-                    <button class="wr-btn" onclick="window.location.href='/mdm/platform-configuration'">Go back to configurations</button>
-                    <button class="wr-btn" onclick="window.location.href='/mdm'">&nbsp;&nbsp;&nbsp;&nbsp;Exit&nbsp;&nbsp;&nbsp;&nbsp;</button>
+                    <button id=btnBack class="wr-btn" onclick="window.location.href='/mdm/platform-configuration'">Go back to configurations</button>
```
Comment (dakshika) [inline modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs]: unquoted attributes are valid on html5 specification, but as a best practice do recommend to use quotes.

File: modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs [hunk 15]
```diff
@@ -472,8 +472,8 @@
...
+                    <button id=btnExit class="wr-btn" onclick="window.location.href='/mdm'">&nbsp;&nbsp;&nbsp;&nbsp;Exit&nbsp;&nbsp;&nbsp;&nbsp;</button>
```
Comment (dakshika) [inline modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs]: unquoted attributes are valid on html5 specification, but as a best practice do recommend to use quotes.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java (hunk 3 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Recommended fixes can be found in PR[1]

[1]. #166

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java (hunk 4 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Recommended fixes can be found in PR[1]

[1]. #166

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java (hunk 7 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java]: Recommended fixes can be found in PR[1]

[1]. #166

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java (hunk 8 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java]: Recommended fixes can be found in PR[1]

[1]. #166

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java (hunk 9 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/platformConfiguration/PlatformConfiguration.java]: Recommended fixes can be found in PR[1]

[1]. #166

File: modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/NotificationValidationTestCase.java (hunk 11 above)
Comment (Kamidu) [inline modules/integration/tests-ui-integration/src/test/java/org/wso2/emm/ui/integration/test/NotificationValidationTestCase.java]: Recommended fixes can be found in PR[1]

[1]. #166

File: modules/apps/jaggery/mdm/src/mdm/units/notification-listing/notification-listing.hbs (hunk 12 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/notification-listing/notification-listing.hbs]: Duly Noted and thank you for the recommendation.
Recommended fixes can be found in PR[1]

[1]. #166

File: modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs (hunk 13 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs]: Duly Noted and thank you for the recommendation.
Recommended fixes can be found in PR[1]

[1]. #166

File: modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs (hunk 14 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs]: Duly Noted and thank you for the recommendation.
Recommended fixes can be found in PR[1]

[1]. #166

File: modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs (hunk 15 above)
Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/platform-configuration/platform-configuration.hbs]: Duly Noted and thank you for the recommendation.
Recommended fixes can be found in PR[1]

[1]. #166

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.wso2.emm.integration.ui.pages.UIElementMapper;
+
+import java.io.IOException;
+
+public class NotificationView {
```
Comment (pamod) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Better have a javadoc with class definition, explaining what the class does

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+    private WebDriver driver;
+    private UIElementMapper uiElementMapper;
+    private static final Log log = LogFactory.getLog(NotificationView.class);
+
+    public NotificationView(WebDriver driver) throws IOException {
```
Comment (pamod) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Also any reason for not including java docs for this method ?

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java (hunk 16 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: Javadoc comments for the class was included in a recent PR.

File: modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java (hunk 17 above)
Comment (Kamidu) [inline modules/integration/tests-common/ui-pages/src/main/java/org/wso2/emm/integration/ui/pages/Notification/NotificationView.java]: These comments were removed due to a suggestion from prabathA pointing out the redundant.
Comment can be found below:
/**
     * Constructor for the NotificationView Class
     * @param driver Instance of the current Web driver
     * @throws IOException driver throws an IOException if failed to load the input URL.
     */
Shall we add this comment again to the constructor?
WDYT?

**Pull request merged**